.github/workflows/cicd.yml

```diff
@@ -30,7 +30,7 @@ concurrency:
   cancel-in-progress: true
 
 permissions:
-  contents: read
+  contents: write # Required to upload performance results.
```

Comment:
This is another reason I'd suggest extracting the perf job into a separate GH workflow after this change.

Reply:
Removed.